examples/pxScene2d/src/pxResource.cpp

```diff
@@ -302,7 +305,7 @@ void pxResource::onDownloadComplete(pxFileDownloadRequest* fileDownloadRequest)
     // Call directly rather than queuing
     ((pxResource*)fileDownloadRequest->getCallbackData())->processDownloadedResource(fileDownloadRequest);
     // Clear download data
-    ((pxResource*)fileDownloadRequest->getCallbackData())->mDownloadRequest = 0;
+    //((pxResource*)fileDownloadRequest->getCallbackData())->mDownloadRequest = 0;
```

Comment:
Multiple objects with multiple threads have pointers to this pxFileDownloadRequest object.  It should be deleted here (and is also possible to be deleted inside of pxFileDownloader::downloadFile() if the pxResource has alread been deleted).  This should be changed to:

pxFileDownloadRequest* downloadRequest = ((pxResource*)fileDownloadRequest->getCallbackData())->mDownloadRequest;
delete downloadRequest;
((pxResource*)fileDownloadRequest->getCallbackData())->mDownloadRequest = 0;